src/main/java/org/opentripplanner/model/TripPattern.java

```diff
@@ -161,20 +161,52 @@ public void setHopGeometriesFromPattern(TripPattern other) {
         int sizeOfShortestPattern = Math.min(numberOfStops(), other.numberOfStops());
 
         for (int i = 0; i < sizeOfShortestPattern - 1; i++) {
-            if (other.getHopGeometry(i) != null
-                && other.getStop(i).equals(this.getStop(i))
-                && other.getStop(i + 1).equals(this.getStop(i + 1))) {
+
+            boolean stopsAreSame = other.getStop(i).equals(this.getStop(i)) && other.getStop(i + 1)
+                    .equals(this.getStop(i + 1));
+            boolean parentStationsAreSame = other.getStop(i)
+                    .getParentStation()
```

Comment:
`.getParentStation()` might return null

Reply:
I've change code to use Objects.equals instead so that it is protected against nulls.